### DISCUSSION TITLE
Read and Write options missing

### DISCUSSION THREAD

**Discussion by borjasor** [pgmpy/pgmpy#1758] [Q&A] [answered]
When following the read and write examples in this notebook: https://github.com/pgmpy/pgmpy_notebook/blob/master/notebooks/9.%20Reading%20and%20Writing%20from%20pgmpy%20file%20formats.ipynb

I noticed that ProbModelXMLReader is no longer available from pgmpy.readwrite

and XMLBeliefNetwork does have issues with methods 'set_variables':
```
XMLBeliefNetwork.py:378 in set_variables
    "TYPE": data[var]["TYPE"],

KeyError: 'TYPE'
```
and 'set_distributions':
```
XMLBeliefNetwork.py:430 in set_distributions
    distributions, "DIST", attrib={"TYPE": self.model.nodes[var]["TYPE"]}

KeyError: 'TYPE'
```
also XMLBeliefNetwork does not seem to have any method to write to file such as 'write_file'.

Currently, it seems that the only working read and write format is BIF.

**Comment by ankurankan** [comment, marked as answer]
@borjasor ProbModelXML was removed a while back, so the notebook does need to be updated. I have fixed the error you were getting with XMLBeliefNetwork, so that should work now. In general, the BIF is the most widely used file format and is also the file format implementation that has been tested most thoroughly in pgmpy.
